extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -12,15 +12,109 @@ import * as constants from '../common/constants';
+				return;
+			}
+			selectedBindingType = selectedBinding.type;
+			propertyBag.bindingType = selectedBindingType;
+			TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.startCreateAzureFunctionWithSqlBinding)
+				.withAdditionalProperties(propertyBag).send();
+
+			// prompt user for connection profile to get connection info
+			quickPickStep = 'getConnectionInfo';
+			connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+			if (!connectionInfo) {
+				// User cancelled
+				return;
+			}
+			TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.startCreateAzureFunctionWithSqlBinding)
+				.withAdditionalProperties(propertyBag).withConnectionInfo(connectionInfo).send();
+
+			// list databases based on connection profile selected
+			connectionURI = await vscodeMssqlApi.connect(connectionInfo);
+			listDatabases = await vscodeMssqlApi.listDatabases(connectionURI);
+			const selectedDatabase = (await vscode.window.showQuickPick(listDatabases, {
+				canPickMany: false,
+				title: constants.selectDatabase,
+				ignoreFocusOut: true
+			}));
+
+			if (!selectedDatabase) {
+				// User cancelled
+				return;
+			}
+			connectionInfo.database = selectedDatabase;
+
+			// prompt user for object name to create function from
+			objectName = await azureFunctionsUtils.promptForObjectName(selectedBinding.type);
+
+		} catch (e) {
+			void vscode.window.showErrorMessage(utils.getErrorMessage(e));
+			propertyBag.quickPickStep = quickPickStep;
+			exitReason = 'error';
+			TelemetryReporter.createErrorEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.exitCreateAzureFunctionQuickpick, undefined, utils.getErrorType(e))
+				.withAdditionalProperties(propertyBag).send();
+		}
+	} else {
+		quickPickStep = 'launchFromTable';
+		connectionInfo = node.connectionInfo;
+		// set the database containing the selected table so it can be used
+		// for the initial catalog property of the connection string
+		let newNode: ITreeNodeInfo = node;
+		while (newNode) {
+			if (newNode.nodeType === 'Database') {
+				connectionInfo.database = newNode.metadata.name;
+				break;
+			} else {
+				newNode = newNode.parentNode;
+			}
+		}
+		// Ask binding type for promptObjectName
+		quickPickStep = 'getBindingType';
+		let selectedBinding = await azureFunctionsUtils.promptForBindingType();
+
+		if (!selectedBinding) {
+			// User cancelled
+			return;
+		}
+		selectedBindingType = selectedBinding.type;
+		propertyBag.bindingType = selectedBinding.type;
+		TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.startCreateAzureFunctionWithSqlBinding)
+			.withAdditionalProperties(propertyBag).withConnectionInfo(connectionInfo).send();
+
+		objectName = utils.generateQuotedFullName(node.metadata.schema, node.metadata.name);
+	}
+	if (!objectName || !connectionInfo || !selectedBindingType) {
```

Comment:
I would generally suggest instead that you do these checks right after prompting for each of these. As mentioned above it's better practice to exit as early as possible in those scenarios.